### ISSUE TITLE
`test_skymax` failing with Numpy 2

### ISSUE THREAD

**Issue by titodalcanton** [gwastro/pycbc#4990]
It looks like CI tests have started picking up Numpy 2, and they fail with this error:
```
======================================================================
ERROR: test_filtering (__main__.skymax_test_maker.<locals>.Test.test_filtering)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tito/projects/pycbc/pycbc/test/test_skymax.py", line 376, in test_filtering
    det_stat_prec = compute_max_snr_over_sky_loc_stat\
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tito/.conda/envs/py312-numpy2/lib/python3.12/site-packages/pycbc/filter/matchedfilter.py", line 540, in compute_max_snr_over_sky_loc_stat
    idx_p, _ = events.threshold_only(hplus[analyse_slice],
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tito/.conda/envs/py312-numpy2/lib/python3.12/site-packages/pycbc/scheme.py", line 221, in _scheming_function
    return _import_cache[mgr.state][func](*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tito/.conda/envs/py312-numpy2/lib/python3.12/site-packages/pycbc/events/threshold_cpu.py", line 49, in threshold_inline
    arr = numpy.array(series.data, copy=False, dtype=numpy.complex64)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Unable to avoid copy while creating an array as requested.
If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.
```
This specific error though is just me running `python test/test_skymax.py` from a Conda env with Numpy 2.1.2.

**Comment by titodalcanton** [comment]
I confirm the same command works from a Conda env with Numpy 1.26.4.

**Comment by spxiwh** [comment]
Oh no, this looks like it's probably my fault then if it's in this test!

**Comment by spxiwh** [comment]
I'll try and fix this, but this is almost certainly because I've not used the right `dtype`s in the test.

**Comment by titodalcanton** [comment]
Fixed by #4991.
